### PR TITLE
Refactor the CFTimeZone initialization

### PR DESCRIPTION
**Pull request by compnerd**
Refactor the initialization codepath a bit to make it more obvious how to modify this to support Windows and alter to query the information from the system.

### CODE REVIEW

**Comment by millenomi** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

File: CoreFoundation/CMakeLists.txt [hunk 1]
```diff
@@ -315,6 +316,12 @@ add_framework(CoreFoundation
                 URL.subproj/CFURLComponents.c
                 URL.subproj/CFURLComponents_URIParser.c
                 URL.subproj/CFURLSessionInterface.c)
+if(CMAKE_SYSTEM_NAME STREQUAL Windows)
+  # NOTE(compnerd) this is a workaround for an empty include directory being
```
[the remark this answers is not in the document]
Comment (compnerd) [reply CoreFoundation/CMakeLists.txt]: Naw, I figured out how to fix it instead.

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by spevans** [comment]
@swift-ci please test

**Comment by millenomi** [comment]
@swift-ci please test and merge

**Comment by compnerd** [comment]
@swift-ci please test and merge

**Pull request merged**